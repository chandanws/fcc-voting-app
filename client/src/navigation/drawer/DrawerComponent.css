.drawer {
  position: absolute;
  top: 8rem;
  left: 0;
  width: 50vw;
  height: calc(100vh - 8rem);
  transition: left 0.5s;
  z-index: 10;
  background: white;
  border: 1px solid black;
  border-top: 0;
}

.drawer--hidden {
  left: -60vw;
}

.not-drawer {
  width: 50vw;
  position: absolute;
  top: 8rem;
  height: calc(100vh - 8rem);
  left: 50vw;
}

.not-drawer--hidden {
  left: -60vw;
}
