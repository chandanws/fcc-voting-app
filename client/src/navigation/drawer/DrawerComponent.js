import React from "react";
import "./DrawerComponent.css";
import { useModifierWithBlock } from "../../helpers/helpers";
const DrawerComponent = props => {
  const modifiers = props.drawerOpen ? [] : ["hidden"];
  const classes = useModifierWithBlock("drawer", modifiers);
  return <div className={classes}>Drawer Component</div>;
};

export default DrawerComponent;
