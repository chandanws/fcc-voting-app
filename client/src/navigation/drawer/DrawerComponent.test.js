import React from "react";
import DrawerComponent from "./DrawerComponent";
import { shallow } from "enzyme";

describe("<DrawerComponent />", () => {
  it("should render correclty", () => {
    const wrapper = shallow(<DrawerComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
