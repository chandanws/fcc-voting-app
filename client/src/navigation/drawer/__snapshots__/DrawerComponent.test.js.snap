// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerComponent /> should render correclty 1`] = `
<div
  className="drawer-wrapper"
>
  <div
    className="drawer-base drawer drawer--hidden"
  >
    <List
      modifiers={Array []}
    >
      <Component
        modifiers={Array []}
      >
        <Button
          className=""
          component="button"
          modifiers={Array []}
        >
          Profile
        </Button>
      </Component>
      <Component
        modifiers={Array []}
      >
        <Button
          className=""
          component="button"
          modifiers={Array []}
        >
          New
        </Button>
      </Component>
      <Component
        modifiers={Array []}
      >
        <Button
          className=""
          component="button"
          modifiers={Array []}
        >
          Logout
        </Button>
      </Component>
    </List>
  </div>
  <div
    className="drawer-base not-drawer not-drawer--hidden"
  />
</div>
`;
